pkg/apis/infra/v1alpha1/cluster.go

```diff
@@ -51,6 +51,13 @@ type CustomClusterSpec struct {
 
 	// CNIConfig is the configuration for the CNI of the cluster on VMs.
 	CNI CNIConfig `json:"cni"`
+
+	// KubesprayVersion is the version of Kubespray used by the customCluster manager worker pod.
+	KubesprayVersion string `json:"kubesprayVersion"`
+
+	// UpgradeK8sVersion represents the target version of Kubernetes that we desire to upgrade to.
+	// if UpgradeK8sVersion existed and is different to kcp.spec.version, then Kurator will reconcile to ensure that the provisioned cluster's version upgrade to upgradeK8sVersion.
+	UpgradeK8sVersion string `json:"upgradeK8sVersion,omitempty"`
```

Comment:
We have one filed in KCP, IMO we should judge if that field updated

Reply:
At the beginning, I planned to use the 'kcp.spc.version' as the variable which set by user for upgrading, but kcp has its own restrictions that cannot be configured more than one minor version difference(you can not config ‘kcp.spc.version’ from 1.22.0 to 1.24.0 because of the validator of kcp), so I created a variable named 'upgradeK8sVersion'  in customCluster. Of course,  in current situation, after the provisioned cluster updated, it will also be synchronized to kcp

Reply:
Can you explain why 1.22->1,24 is not allowed

Reply:
https://github.com/kubernetes-sigs/cluster-api/blob/75afbcb82cc9c5599edb17ff599063fa5765a93d/controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_webhook.go#L615-L631

Reply:
Confirm with @zirain

Reply:
due to https://kubernetes.io/releases/version-skew-policy/

Reply:
Should look into kubespray upgrade policy. I think it must be different from cluster-api

Reply:
kubespray use kubeadm to upgrade and also "Skipping MINOR versions when upgrading is unsupported". https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade/     I attempted to upgrade from version 1.22.0 to 1.24.0 using the kubespay job, but encountered an error from kubeadm.

Reply:
Now i am super sure we donot need another such field